imu_sensor_broadcaster/src/imu_sensor_broadcaster.cpp

```diff
@@ -110,22 +120,75 @@ controller_interface::CallbackReturn IMUSensorBroadcaster::on_deactivate(
   return CallbackReturn::SUCCESS;
 }
 
-controller_interface::return_type IMUSensorBroadcaster::update(
+controller_interface::return_type IMUSensorBroadcaster::update_and_write_commands(
   const rclcpp::Time & time, const rclcpp::Duration & /*period*/)
 {
-  if (realtime_publisher_ && realtime_publisher_->trylock())
+  imu_sensor_->get_values_as_message(state_message_);
+  doTransform(state_message_, state_message_, r_);
```

Comment:
```suggestion
  sensor_msgs::msg::Imu input_imu;
  imu_sensor_->get_values_as_message(input_imu);
  doTransform(input_imu, r_, state_message_);
```

Should we do it this way?. No extra memory is allocated this way?. I'm a bit skeptic about using same variable in input and output. It is just me XD